src/backend/distributed/executor/executor.c

```diff
@@ -2086,6 +2195,30 @@ PlacementExecutionDone(TaskPlacementExecution *placementExecution, bool succeede
 }
 
 
+/*
+ * ShouldMarkPlacementsInvalidOnFailures returns true if the failure
+ * should trigger marking placements invalid.
+ */
+static bool
+ShouldMarkPlacementsInvalidOnFailure(DistributedExecution *execution)
+{
+	DistributedPlan *distributedPlan = execution->plan;
+	CmdType operation = distributedPlan->operation;
+
+	if (operation == CMD_SELECT || execution->errorOnAnyFailure)
```

Comment:
It would be nice if we could generalise this a bit. I could see us wanting to push down router queries with modifying CTEs in the near future, which are technically SELECTs. We also don't want to make placements as inactive when ANALYZE fails.

In general, it seems we need more a reliable version of `IsModifyDistributedPlan` that can distinguish which plans modify the database.